packages/contracts/src/FluidLocker.sol

```diff
@@ -441,40 +712,272 @@ contract FluidLocker is Initializable, ReentrancyGuard, IFluidLocker {
     }
 
+    /**
+     * @notice Swaps ETH for SUP tokens using Uniswap V3 (Pumponomics)
+     * @param ethx ETHx address
+     * @param ethAmount The amount of ETH to swap
+     */
+    function _pump(address ethx, uint256 ethAmount) internal {
+        IERC20(ethx).approve(address(SWAP_ROUTER), ethAmount);
+
+        // No need slippage protection here as it is
+        // implicitely covered by the `_createPosition` slippage protection
+        IV3SwapRouter.ExactInputSingleParams memory swapParams = IV3SwapRouter.ExactInputSingleParams({
+            tokenIn: ethx,
+            tokenOut: address(FLUID),
+            fee: ETH_SUP_POOL.fee(),
+            recipient: address(this),
+            amountIn: ethAmount,
+            amountOutMinimum: 0,
+            sqrtPriceLimitX96: 0
+        });
+
+        SWAP_ROUTER.exactInputSingle(swapParams);
+    }
+
+    /**
+     * @notice Creates a new Uniswap V3 position with the specified amounts of tokens
+     * @param ethAmount The desired amount of ETH to add as liquidity
+     * @param supAmount The desired amount of SUP to add as liquidity
+     * @return positionTokenId The token identifier of the newly created position
+     * @return depositedEthAmount The actual amount of ETH deposited as liquidity
+     * @return depositedSupAmount The actual amount of SUP deposited as liquidity
+     */
+    function _createPosition(uint256 ethAmount, uint256 supAmount)
+        internal
+        returns (uint256 positionTokenId, uint256 depositedEthAmount, uint256 depositedSupAmount)
+    {
+        bool zeroIsSup = ETH_SUP_POOL.token0() == address(FLUID);
+
+        INonfungiblePositionManager.MintParams memory mintParams = _formatMintParams(zeroIsSup, ethAmount, supAmount);
+
+        // Create the UniswapV3 position
+        (uint256 tokenId, uint128 liquidity, uint256 depositedAmount0, uint256 depositedAmount1) =
+            NONFUNGIBLE_POSITION_MANAGER.mint(mintParams);
+
+        // Set the tax free withdraw timestamp
+        taxFreeExitTimestamps[tokenId] = block.timestamp + TAX_FREE_WITHDRAW_DELAY;
+
+        // Update the aggregated liquidity balance
+        _liquidityBalance += liquidity;
+
+        if (!FLUID.isMemberConnected(address(LP_DISTRIBUTION_POOL), address(this))) {
+            // Connect this locker to the LP Tax Distribution Pool
+            FLUID.connectPool(LP_DISTRIBUTION_POOL);
+        }
+
+        // Update the liquidity provider units
+        STAKING_REWARD_CONTROLLER.updateLiquidityProviderUnits(_liquidityBalance);
+
+        (depositedSupAmount, depositedEthAmount) = _sortOutAmounts(zeroIsSup, depositedAmount0, depositedAmount1);
+
+        positionTokenId = tokenId;
+    }
+
+    /**
+     * @notice Decreases liquidity from a Uniswap V3 position
+     * @param tokenId The ID of the NFT position to decrease liquidity from
+     * @param liquidityToRemove The amount of liquidity to remove from the position (only collect fees if set to 0)
```

Comment:
`only collect fees if set to 0` could be misunderstood as: the method only collects fees if the param is set to 0.
I'd add this to the notice: "... and collects fees".